cmds/core/dhclient/dhclient.go

```diff
@@ -47,51 +47,79 @@ var (
 	v4Port = flag.Int("v4-port", dhcpv4.ServerPort, "DHCPv4 server port to send to")
 )
 
-func main() {
-	flag.Parse()
-	if len(flag.Args()) > 1 {
-		log.Fatalf("only one re")
-	}
+type opts struct {
+	timeout  int
+	retry    int
+	dryRun   bool
+	verbose  bool
+	vverbose bool
+	ipv4     bool
+	ipv6     bool
+	v6Port   int
+	v6Server string
+	v4Port   int
+}
 
-	if len(flag.Args()) > 0 {
-		ifName = flag.Args()[0]
+func run(args []string, opts opts) error {
+	var ifName string
+	switch len(args) {
+	case 0:
+		ifName = "^e.*"
+	case 1:
+		ifName = args[0]
+	default:
+		return fmt.Errorf("more than one interface specified")
 	}
 
 	filteredIfs, err := dhclient.Interfaces(ifName)
 	if err != nil {
-		log.Fatal(err)
+		return err
+	}
+
+	if err := configureAll(filteredIfs, opts); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+	opts := opts{timeout: *timeout, retry: *retry, dryRun: *dryRun, verbose: *verbose, vverbose: *vverbose, ipv4: *ipv4, ipv6: *ipv6, v6Port: *v6Port, v6Server: *v6Server, v4Port: *v4Port}
+	if err := run(args, opts); err != nil {
+		log.Fatalf("dhclient: %v", err)
 	}
 
-	configureAll(filteredIfs)
 }
 
-func configureAll(ifs []netlink.Link) {
-	packetTimeout := time.Duration(*timeout) * time.Second
+func configureAll(ifs []netlink.Link, opts opts) error {
```

Comment:
do you want to consider *opts instead of the struct by value?

Reply:
Sure, by value is less efficient. fixed in the latest commit.